src/UriHelper.php

```diff
@@ -160,13 +160,15 @@ public static function decomposeUri(UriInterface $uri)
         $pathPattern = '(?<hash>-.+-)';
         $path = $uri->getPath();
         // hash with seo-filename
-        if (preg_match('#^/'.$stackPattern.'/'.$hashPattern.'/'.$filenamePattern.'\.'.$formatPattern.'$#', $path, $matches) ||
+        if (preg_match('#^/'.$stackPattern.'/'.$hashPattern.'/'.$filenamePattern.'\.'.$formatPattern.'$#', $path, $matches)
             // hash without seo-filename
-            preg_match('#^/'.$stackPattern.'/'.$hashPattern.'.'.$formatPattern.'$#', $path, $matches) ||
+            || preg_match('#^/'.$stackPattern.'/'.$hashPattern.'.'.$formatPattern.'$#', $path, $matches)
             // remote_path with seo-filename
-            preg_match('#^/'.$stackPattern.'/'.$pathPattern.'/'.$filenamePattern.'\.'.$formatPattern.'$#', $path, $matches) ||
+            || preg_match('#^/'.$stackPattern.'/'.$pathPattern.'/'.$filenamePattern.'\.'.$formatPattern.'$#', $path, $matches)
             // remote_path without seo-filename
-            preg_match('#^/'.$stackPattern.'/'.$pathPattern.'.'.$formatPattern.'$#', $path, $matches)) {
+            || preg_match('#^/'.$stackPattern.'/'.$pathPattern.'.'.$formatPattern.'$#', $path, $matches)
+        ) {
+            // @phpstan-ignore-next-line
```

Comment:
we should upgrade to phpstan v1.11 and ignore specific errors here (they introduced error identifiers in v1.11)

Reply:
a, nice. and with `phpstan -v` one gets to see the rules